### ISSUE TITLE
Lint x != _ || x != _

### ISSUE THREAD

**Issue by llogiq** [Manishearth/rust-clippy#853]
Also probably `x == _ && x == _`. This is either a duplicate (if both `_` are equal) or always false (true for the former).

**Comment by oli-obk** [comment]
we could probably let the `logic_bug` lint catch this. If the `_` are equal, it's already caught. If they differ, we can add some logic to merge them into the same terminal and see if that eliminates any terminals.

**Comment by felix91gr** [comment]
What happened to this lint? Did it become part of `logic_bug`, @oli-obk?

**Comment by oli-obk** [comment]
nope. It was never integrated. We need to be careful which values we convert to the same terminal. E.g. `x == foo() && x == bar()` may easily be true, this trick should only be applied if the `_` is a literal.
